kubetest/anywhere.go

```diff
@@ -237,13 +239,32 @@ func (k *kubernetesAnywhere) IsUp() error {
 }
 
 func (k *kubernetesAnywhere) DumpClusterLogs(localPath, gcsPath string) error {
-	// TODO(pipejakob): the default implementation (log-dump.sh) doesn't work for
-	// kubernetes-anywhere yet, so just skip attempting to dump logs.
-	// https://github.com/kubernetes/kubeadm/issues/256
-	log.Print("DumpClusterLogs is a no-op for kubernetes-anywhere deployments. Not doing anything.")
-	log.Print("If you care about enabling this feature, follow this issue for progress:")
-	log.Print("    https://github.com/kubernetes/kubeadm/issues/256")
-	return nil
+	if !*kubernetesAnywhereDumpClusterLogs {
+		log.Printf("Cluster log dumping disabled for Kubernetes Anywhere.")
+		return nil
+	}
+	logDumpPath := "./cluster/log-dump/log-dump.sh"
+	// cluster/log-dump/log-dump.sh only exists in the Kubernetes tree
+	// post-1.3. If it doesn't exist, print a debug log but do not report an error.
```

Comment:
We don't need to consider anything below v1.6.
When v1.9 comes out support is dropped for v1.6, etc.